### DISCUSSION TITLE
Best way to run through multiple hosts?  Why does xpra ignore --pulseaudio=no?

### DISCUSSION THREAD

**Discussion by AGI-chandler** [Xpra-org/xpra#3708] [Q&A] [answered]
I'm trying to optimize how xpra is run and I don't need sound support so I'm specifying `--pulseaudio=no` but then I still see:
```
 audio playback using pulseaudio device:
 audio playback  'Built-in Audio'
 audio playback using 'opus' audio codec
```
Why's that?

**Comment by totaam** [comment]
As per the man page, the `pulseaudio` switch controls whether a pulseaudio server is started or not.
What you are seeing there is the audio playback - probably on the client I would guess.

> I'm trying to optimize how xpra is run

It's not clear to me what you're trying to "optimize".

**Comment by AGI-chandler** [comment]
Thanks, but even with your's and the manual's explanation of that switch, seems to me there just shouldn't be any pulseaudio running.  There doesn't really need to be any audio support for this session.  I'm trying to optimize by removing unneeded passthrough options such as that, because all I really need to run is firefox for a short period.

Also the computer I need to run firefox on is actually a third, with a second computer between us.  So that's why the optimization is important.  What would be the best way to run firefox like this?  The old way was to ssh to the second computer, then ssh to the third computer and start firefox with X11forwarding... painfully slow, pretty much unusable.  However, xpra has been crashing the session it seems... I've tried running xpra on my computer and starting the server on the second computer via ssh, then opened an xterm and ssh to the third computer to start firefox on command line.  I've also tried just ssh'ing to the second computer, then run xpra there to start the server on the third computer, with the X11forwarding xpra tray icon shows up on my computer with some lag.  In both cases firefox will start but after a short while the xpra server will just shutdown.

**Comment by totaam** [comment, marked as answer]
> seems to me there just shouldn't be any pulseaudio running
> There doesn't really need to be any audio support for this session

Then use `--speaker=no --microphone=no`.

> I'm trying to optimize by removing unneeded passthrough options such as that, because all I really need to run is firefox for a short period

"optimize" can mean so many different things, especially with tools like xpra.
I assume that here you mean removing things you don't need.

> However, xpra has been crashing the session it seems...
Unlikely but possible. What did your server log show?

> In both cases firefox will start but after a short while the xpra server will just shutdown
That seems weird.
`firefox` in xpra is used a lot.
Please see https://github.com/Xpra-org/xpra/wiki/Reporting-Bugs

> I've also tried just ssh'ing ...
There are many ways of doing this.
Whatever you do, use xpra end-to-end, never use xpra together with remote-X11 like `ssh -X`.
Just use the 2-hop ssh: #2041
You can also add your target host to your `.ssh/config` via a `ProxyCommand`.

**Reply by AGI-chandler** [reply]
> Then use `--speaker=no --microphone=no`.

Thanks


> I assume that here you mean removing things you don't need.

Yes


> Just use the 2-hop ssh: #2041

At first I only found this option using the GUI.  Not clear how to make it work there.  It seems to operate outside the normal `~/.ssh/config` as it doesn't recognize my short `host` names nor the identities/keys I have saved.  When I tried using the full hostname in xpra I was asked to enter the password to the remote host, which I don't know off hand.  Either way, it should be using the keypair to authenticate...

Then I found the way to run it on the command line which is preferred anyway, xpra start `ssh://3rdComputer/?proxy=ssh://2ndComputer`

Thanks!


> You can also add your target host to your `.ssh/config` via a `ProxyCommand`.

This works too, using newer `ProxyJump` option. Thanks

**Comment by totaam** [comment]
See also https://github.com/Xpra-org/xpra/issues/3835
